org.jacoco.core/src/org/jacoco/core/matcher/Matcher.java

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Jeffry Gaston - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.matcher;
+
+public interface Matcher<T> {
+	boolean matches(final T s);
+}
```

Comment:
@mathjeff For our artifacts that themselve can be used as dependencies (i.e. core - everything except Maven Plugin and CLI) I would prefer to avoid any dependency on any third-party artifact except ASM to avoid any questions about compatibility and transitive dependencies. And so especially avoid dependency on Guava (https://dev.eclipse.org/mhonarc/lists/cross-project-issues-dev/msg14431.html). Also recall that JaCoCo can still be executed under Java 5, while latest versions of Guava and Apache commons-collections, commons-lang, commons-io can't. Also recall that for integration into EclEmma any JaCoCo dependency should be in Eclipe Orbit. Shading of dependencies raises licensing/redistribution questions. So all in all - IMO a way simpler to just avoid them. A single interface definitely not worth the efforts.